### ISSUE TITLE
feature: --split-pages option: split pages in group

### ISSUE THREAD

**Issue by Mark-Joy** [qpdf/qpdf#616]
As per described in #30: 
```
qpdf in.pdf in.pdf --pages in.pdf 1-5,11,60,70-100 -- --split-pages=3
``` 
will split the pdf into: 
```
in-1.pdf: page 1,2,3
in-2.pdf: page 4,5,11
in-3.pdf: page 60,70,71
in-4.pdf: page 72,73,74
in-5.pdf: page 75,76,77
...
```
Could you please change it to generate the below: 
```
in-1.pdf: page 1,2,3
in-2.pdf: page 4,5
in-3.pdf: page 11
in-4.pdf: page 60
in-5.pdf: page 70,71,72
...
```
In other words, it split pages in group seperated by comma`,`.
Better if it can name the output file accordingly: 
```
in-1-3: page 1,2,3
in-4-5: page 4,5
in-11: page 11
in-60: page 60
in-70-72: page 70,71,72
...
```

Thank you.

**Comment by jberkenbilt** [comment]
Thanks for the idea, but I'm sorry to say that I am not going to consider making this change.

This would be an incompatible change, and the way the numeric range is specified is discarded during command-line argument parsing. In addition to not being a compatible change, this change would make different specifications of equivalent page ranges behave differently, which I think would be surprising and counterintuitive behavior. I think, for your use case, you'll just have to run multiple qpdf commands. You could use a separate `--pages` specification for each group of pages that you want.

It would be possible to come up with a completely different syntax for creating individual groups of pages based on separate ranges, but I think it's probably just cluttering up the API given that you can already do it by running qpdf multiple times.

**Comment by Mark-Joy** [comment]
Yes, you are right. This would easily be solved with a shell script.
However, considering user view, it's best to solve it by `qpdf` syntax, running in 1 go.
The `--split-pages` option has not much use in my case. My intention was to split pdf in 1 go.
Example: 
`qpdf in.pdf in.pdf --pages in.pdf 1-5,11,60,70-100`
would split pdf into in-1-5.pdf, in-11.pdf, in-60.pdf and in-70-100.pdf
I think this has more use cases than `--split-pages`, but I could certainly be wrong.

**Comment by jberkenbilt** [comment]
Okay, I like the idea of being able to split in named groups, so I'll leave this open and add the label I use for page-related features. At some point, I am going to add lots of other ways to split, such as splitting based on outlines or other document-level features, and this feature might fit will into that set of modifications. I suspect it will be months before I tackle that pile of issues. If implemented, it will be by adding a new option, not changing the semantics of `--split-pages`, for reasons I explained. However I think your main point is the need to split in named groups, not to change the behavior of `--split-pages` per se.

> I think this has more use cases than `--split-pages`, but I could certainly be wrong.

I don't (and can't really) gather metrics on what features are used, but I know that `--split-pages` is a heavily used feature that was requested multiple times before I added it.

Thanks for the report and suggestion!

**Comment by Perangelot** [comment]
> Yes, you are right. This would easily be solved with a shell script. However, considering user view, it's best to solve it by `qpdf` syntax, running in 1 go. The `--split-pages` option has not much use in my case. My intention was to split pdf in 1 go. Example: `qpdf in.pdf in.pdf --pages in.pdf 1-5,11,60,70-100` would split pdf into in-1-5.pdf, in-11.pdf, in-60.pdf and in-70-100.pdf I think this has more use cases than `--split-pages`, but I could certainly be wrong.

The command does not work for me. It tells me that the input file is defined twice and that `--` is missing at the end. If I change that, I do not get in-1-5.pdf, in-11.pdf, in-60.pdf and in-70-100.pdf, but a single file with these pages. I need the separate files, though. Any idea?

**Comment by jberkenbilt** [comment]
@Perangelot That command was a suggestion. That's now how qpdf actually works. You can find more detail in the other comments.

**Comment by Perangelot** [comment]
Is it _now_ or _not_ how qpdf actually works? I have looked at the other comments. Have I understood correctly that it is not possible to extract one pdf file per page range in one command?

**Comment by jberkenbilt** [comment]
Sorry, _not_. That's the worst typo in the English language. It is _not_ how it works, and it is not how it is intended to work. You are correct that there is no way to extract one file per range. The behavior is dependent on the set of pages in the range, and not how the range is specified. That won't change since changing it would break compatibility and also require a rework of how the whole page range thing is handled internally. (The internal representation does not preserve the way the range was specified.)

**Comment by jberkenbilt** [comment]
Some form of this will be done. See also #741. I'm working through the issues with `pages` on them now and making lists of specific features. I don't know what the syntax will be yet.

**Comment by jberkenbilt** [comment]
This issue is part of the qpdf pages epic. If you are interested in following, please see #1104.
